internal/manifest/statecheck/state_check.go

```diff
@@ -0,0 +1,78 @@
+package statecheck
+
+import (
+	"context"
+
+	apiappsv1 "k8s.io/api/apps/v1"
+	"k8s.io/cli-runtime/pkg/resource"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	declarativev2 "github.com/kyma-project/lifecycle-manager/internal/declarative/v2"
+)
+
+type ManagerStateCheck struct {
+	statefulSetChecker     StatefulSetStateChecker
+	deploymentStateChecker DeploymentStateChecker
+}
+
+type ManagerKind string
+
+const (
+	DeploymentKind  ManagerKind = "Deployment"
+	StatefulSetKind ManagerKind = "StatefulSet"
+)
+
+type Manager struct {
+	Kind ManagerKind
+	*apiappsv1.Deployment
+	*apiappsv1.StatefulSet
+}
+
+func NewManagerStateCheck() *ManagerStateCheck {
+	return &ManagerStateCheck{
+		statefulSetChecker:     NewStatefulSetStateCheck(),
+		deploymentStateChecker: NewDeploymentStateCheck(),
+	}
+}
```

Comment:
Please inject these dependencies from the outside. Then we can also easily create a test file state_check.go. We can supply test stubs instead of the real dependencies. Then we can test that the correct stub is called depending on whether we are passing a deployment or statefulset to GetState. 
We can also do that as a pair programming session if you like.